packages/firestore/src/model/mutation.ts

```diff
@@ -600,22 +579,29 @@ export class TransformMutation extends Mutation {
     });
   }
 
-  get isIdempotent(): boolean {
+  extractBaseValue(maybeDoc: MaybeDocument | null): ObjectValue | null {
+    let baseObject: ObjectValue | null = null;
     for (const fieldTransform of this.fieldTransforms) {
-      if (!fieldTransform.isIdempotent) {
-        return false;
+      const existingValue =
+        maybeDoc instanceof Document
+          ? maybeDoc.field(fieldTransform.field)
+          : undefined;
+      const coercedValue = fieldTransform.transform.computeBaseValue(
```

Comment:
Not a comment but a question: where is the implementation of `computeBaseValue` defined. I cannot find it..

Reply:
It's added in this PR: https://github.com/firebase/firebase-js-sdk/pull/1928/files#diff-eb38c9da3263f013aac932a74339324dR228

Reply:
Ah, I overlooked..thx.